translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go

```diff
@@ -181,7 +181,20 @@ func (t *TimestampLayout) ApplyRule(input interface{}) (returnKey string, return
 		res := checkAndReplace(val.(string), TimeFormatMap)
 		//If user provide with the specific timestamp_format, use the one that user provide
 		returnKey = "timestamp_layout"
-		returnVal = res
+		input := val.(string)
+		// Go doesn't support _2 option for month in day as a result need to set
+		// timestamp_layout with 2 strings which support %m and %-m
+		if strings.Contains(input, "%m") {
+			input := strings.Replace(input, "%m", "%-m", -1)
+			defaultOption := checkAndReplace(input, TimeFormatMap)
```

Comment:
nit: `defaultOption` doesn't really fit what this is doing. Consider renaming.

Reply:
renamed to `alternativeLayout`